app/assets/javascripts/admin/subscriptions/services/subscription_resource.js.coffee

```diff
@@ -27,5 +27,6 @@ angular.module("admin.subscriptions").factory 'SubscriptionResource', ($resource
   })
 
   angular.extend(resource.prototype, SubscriptionActions)
+  angular.extend(resource.prototype, SubscriptionFunctions)
```

Comment:
So we put the functions into a different class, but then extend the resource prototype. Does that mean that *all* resources have these functions now? Why don't we declare these functions in the SubscriptionResource?

Reply:
Not quite my understanding but I could be wrong. As far as I understand it, the call to `$resource` above returns a `Resource` constructor, which is then used to build new `subscriptions` objects which have access the nice resource api elements ($create, $update, etc). I think I can safely extend the prototype of just this constructor independently of other resource constructors, because constructors are not singletons or global in scope? Here I am extending the prototype of the constructor, to give all the `subscriptions` I instantiate access to the functions defined by `SubscriptionActions` and `SubscriptionFunctions`. I kind of made this up as I went along, so if there is a better way of doing it I am happy to use that instead.

Reply:
By doing it this way I am kind of defining the functions in the SubscriptionResource, but I'm just including them like you would a module in ruby, instead of having them all jumbled into one factory.